transaction/std/src/signed_contingent_input_builder.rs

```diff
@@ -0,0 +1,2272 @@
+                        .transpose()
+                } else {
+                    Ok(None)
+                }
+            },
+            rng,
+        );
+        // Put the memo builder back
+        self.memo_builder = Some(mb);
+        result
+    }
+
+    /// Add an output to the transaction, using `fog_hint_address` to construct
+    /// the fog hint.
+    ///
+    /// This is a private implementation detail, and generally, fog users expect
+    /// that the transactions that they recieve from fog belong to the account
+    /// that they are using. The only known use-case where recipient and
+    /// fog_hint_address are different is when sending change transactions
+    /// to oneself, when oneself is a fog user. Sending the change to the
+    /// main subaddress means that you don't have to hit fog once for the
+    /// main subaddress and once for the change subaddress, so it cuts the
+    /// number of requests in half.
+    ///
+    /// # Arguments
+    /// * `amount` - The amount of this output
+    /// * `recipient` - The recipient's public address
+    /// * `fog_hint_address` - The public address used to create the fog hint
+    /// * `memo_fn` - The memo function to use (see TxOut::new_with_memo)
+    /// * `rng` - RNG used to generate blinding for commitment
+    fn add_output_with_fog_hint_address<RNG: CryptoRng + RngCore>(
+        &mut self,
+        amount: Amount,
+        recipient: &PublicAddress,
+        fog_hint_address: &PublicAddress,
+        memo_fn: impl FnOnce(MemoContext) -> Result<Option<MemoPayload>, NewMemoError>,
+        rng: &mut RNG,
+    ) -> Result<(TxOut, TxOutConfirmationNumber), TxBuilderError> {
+        let (hint, pubkey_expiry) =
+            crate::transaction_builder::create_fog_hint(fog_hint_address, &self.fog_resolver, rng)?;
+
+        let (tx_out, shared_secret) = crate::transaction_builder::create_output_with_fog_hint(
+            self.block_version,
+            amount,
+            recipient,
+            hint,
+            memo_fn,
+            rng,
+        )?;
+
+        let (amount, blinding) = tx_out
+            .masked_amount
+            .get_value(&shared_secret)
+            .expect("TransactionBuilder created an invalid Amount");
+        let output_secret = OutputSecret { amount, blinding };
+
+        self.impose_tombstone_block_limit(pubkey_expiry);
+
+        self.outputs_and_secrets
+            .push((tx_out.clone(), output_secret));
+
+        let confirmation = TxOutConfirmationNumber::from(&shared_secret);
+
+        Ok((tx_out, confirmation))
+    }
+
+    /// Sets the tombstone block, clamping to smallest pubkey expiry value.
+    ///
+    /// # Arguments
+    /// * `tombstone_block` - Tombstone block number.
+    pub fn set_tombstone_block(&mut self, tombstone_block: u64) -> u64 {
+        self.tombstone_block = min(tombstone_block, self.fog_tombstone_block_limit);
+        self.tombstone_block
```

Comment:
IIUC we can just return the result of the assignment
```suggestion
        self.tombstone_block = min(tombstone_block, self.fog_tombstone_block_limit)
```

Reply:
in my understanding, this works in C but not in rust -- the result of assignment is `()`. willing to be wrong

Reply:
yeah, it doesn't build that way. i think this is the reference:

https://doc.rust-lang.org/reference/expressions/operator-expr.html#basic-assignments

I think they got rid of this because they felt that in C and C++ this creates too much cryptic code, like the

```
a = ++b = c++
```
etc.